int8 speed
int8 dir
